frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-page.tsx

```diff
@@ -25,95 +25,128 @@ import { OperatorHubItem, OperatorHubCSVAnnotations, InstalledState } from './in
-          provider: _.get(pkg, 'status.provider.name', _.get(pkg, 'metadata.labels.provider')),
-          providerType: getOperatorProviderType(pkg),
-          tags: [],
-          version: _.get(currentCSVDesc, 'version'),
-          categories: _.get(currentCSVAnnotations, 'categories', '')
-            .split(',')
-            .map((category) => category.trim()),
-          catalogSource: _.get(pkg, 'status.catalogSource'),
-          catalogSourceNamespace: _.get(pkg, 'status.catalogSourceNamespace'),
-          certifiedLevel,
-          healthIndex,
-          repository,
-          containerImage,
-          createdAt,
-          support,
-          capabilityLevel,
-          marketplaceActionText,
-          marketplaceRemoteWorkflow,
-          marketplaceSupportWorkflow,
-          validSubscription: validSubscription ? JSON.parse(validSubscription) : undefined,
-          infraFeatures: infrastructure ? JSON.parse(infrastructure) : undefined,
-        };
-      },
-    );
+  const localItems = _.get(props, 'packageManifest.data', [] as PackageManifestKind[]); // FIXME optional chaining
+  const getPackageStatus = (pkg) => _.get(pkg, 'status'); // FIXME optional chaining and/or is this needed?
+
+  const items: OperatorHubItem[] = React.useMemo(
+    () =>
+      marketplaceItems
+        .concat(localItems)
+        .filter((pkg) => {
+          // if a package does not have status.defaultChannel, exclude it so the app doesn't fail
+          const { defaultChannel } = getPackageStatus(pkg);
+          if (!defaultChannel) {
+            // eslint-disable-next-line no-console
+            console.warn(
+              `PackageManifest ${pkg.metadata.name} has no status.defaultChannel and has been excluded`,
+            );
+            return false;
+          }
+          return true;
+        })
+        .map(
+          (pkg): OperatorHubItem => {
+            const { channels, defaultChannel } = getPackageStatus(pkg);
+            const { currentCSVDesc } = _.find(channels || [], { name: defaultChannel } as any);
+            // FIXME optional chaining
+            const currentCSVAnnotations: OperatorHubCSVAnnotations = _.get(
+              currentCSVDesc,
+              'annotations',
+              {},
+            );
+            const {
+              certifiedLevel,
+              healthIndex,
+              repository,
+              containerImage,
+              createdAt,
+              support,
+              capabilities: capabilityLevel,
+              'marketplace.openshift.io/action-text': marketplaceActionText,
+              'marketplace.openshift.io/remote-workflow': marketplaceRemoteWorkflow,
+              'marketplace.openshift.io/support-workflow': marketplaceSupportWorkflow,
+              'operators.openshift.io/infrastructure-features': infrastructure,
+              'operators.openshift.io/infrastructure-features': validSubscription,
```

Comment:
Are we parsing correct annotation here line 76?

Reply:
No, oops, I fixed that but forgot to push it.